src/test/scala/integration/TopicLoaderIntSpec.scala

```diff
@@ -190,25 +190,87 @@ class TopicLoaderIntSpec extends KafkaSpecBase[IO] {
 
       val (preLoad, postLoad) = records(1 to 15).splitAt(10)
 
+      def testState(loadState: Ref[IO, Boolean], topicState: Ref[IO, Seq[(String, String)]]): IO[Assertion] =
+        loadAndRunR(NonEmptyList.one(testTopic1))(
+          _ => loadState.set(true),
+          r => topicState.getAndUpdate(_ :+ r).void
+        ).surround {
+          for {
+            _         <- eventually(topicState.get.asserting(_ should contain theSameElementsAs preLoad))
+            _         <- eventually(loadState.get.asserting(_ shouldBe true))
+            _         <- publishStringMessages(testTopic1, postLoad)
+            assertion <-
+              eventually(
+                topicState.get.asserting(_ should contain theSameElementsAs (preLoad ++ postLoad))
+              )
+          } yield assertion
+        }
+
       for {
         loadState  <- Ref.of[IO, Boolean](false)
         topicState <- Ref.empty[IO, Seq[(String, String)]]
         _          <- createCustomTopics(NonEmptyList.one(testTopic1))
         _          <- publishStringMessages(testTopic1, preLoad)
-        assertion  <- loadAndRunR(NonEmptyList.one(testTopic1))(
-                        _ => loadState.set(true),
-                        r => topicState.getAndUpdate(_ :+ r).void
-                      ).surround {
-                        for {
-                          _         <- eventually(topicState.get.asserting(_ should contain theSameElementsAs preLoad))
-                          _         <- loadState.get.asserting(_ shouldBe true)
-                          _         <- publishStringMessages(testTopic1, postLoad)
-                          assertion <-
-                            eventually(
-                              topicState.get.asserting(_ should contain theSameElementsAs (preLoad ++ postLoad))
-                            )
-                        } yield assertion
-                      }
+        assertion  <- testState(loadState, topicState)
+      } yield assertion
+    }
+
+    "execute callback if the topic is empty and keep streaming" in withKafkaContext { ctx =>
+      import ctx.given
+
+      val postLoad = records(1 to 15)
+
+      def testState(loadState: Ref[IO, Boolean], topicState: Ref[IO, Seq[(String, String)]]): IO[Assertion] =
```

Comment:
maybe `assertPostLoadRecordsConsumed` if i'm understanding it right

also, can we extract and reuse in the next test? or are they too different

Reply:
https://github.com/sky-uk/fs2-kafka-topic-loader/pull/45/commits/78e1175c36a6e9d3fa187ddadb05501139137db5